docs/articles/persistence/custom-persistence-provider.md

```diff
@@ -0,0 +1,467 @@
+
+In order to support this, we will have to be able to stash all incoming messages while we're creating the tables in the event source. Here's an implementation that we use in our example code:  
+
+[!code-csharp[Startup](../../../src/examples/Akka.Persistence.Custom/Journal/SqliteJournal.cs?name=Startup "Journal startup sequence")]
+
+## Implementing Akka.Persistence SnapshotStore
+
+### Required Method Implementations
+
+#### LoadAsync
+
+```c#
+Task<SelectedSnapshot> LoadAsync(string persistenceId, SnapshotSelectionCriteria criteria)
+```
+
+* **`persistenceId`**: Identification of the persistent actor
+* **`criteria`**: Selection criteria for event loading
+
+This call is protected with a circuit-breaker.
+
+Asynchronously loads a snapshot.
+
+##### Code Sample
+
+`SelectSnapshotSql` in this example refers to this SQL query statement:
+
+[!code-csharp[SelectSnapshotSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=SelectSnapshotSql "SelectSnapshotSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[LoadAsync](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=LoadAsync "LoadAsync method implementation")]
+
+#### SaveAsync
+
+```c#
+Task SaveAsync(SnapshotMetadata metadata, object snapshot)
+```
+
+* **`metadata`**: Snapshot metadata
+* **`snapshot`**: The snapshot object instance to be stored
+
+This call is protected with a circuit-breaker
+
+Asynchronously saves a snapshot.
+
+##### Code Sample
+
+`InsertSnapshotSql` in this example refers to this SQL query statement:
+
+[!code-csharp[InsertSnapshotSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=InsertSnapshotSql "InsertSnapshotSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[SaveAsync](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=SaveAsync "SaveAsync method implementation")]
+
+#### DeleteAsync
+
+```c#
+Task DeleteAsync(SnapshotMetadata metadata);
+```
+
+* **`metadata`**: Snapshot metadata
+
+This call is protected with a circuit-breaker
+
+Deletes the snapshot identified by by the provided `Metadata`
+
+##### Code Sample
+
+`DeleteSnapshotSql` in this example refers to this SQL query statement:
+
+[!code-csharp[DeleteSnapshotSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=DeleteSnapshotSql "DeleteSnapshotSql SQL Statement")]
+
+`DeleteSnapshotRangeSql` in this example refers to this SQL query statement:
+
+[!code-csharp[DeleteSnapshotRangeSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=DeleteSnapshotRangeSql "DeleteSnapshotRangeSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[DeleteAsync](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=DeleteAsync "DeleteAsync method implementation")]
+
+#### DeleteAsync
+
+```c#
```

Comment:
Do we leave this as it is or make use of the range selection though the line number may change in the future?

Reply:
see above